File: 1233. Remove Sub-Folders from the Filesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) 
    {
        sort(begin(folder), end(folder));
        vector<string>ans(1);
        ans[0] = folder[0];

        int n = folder.size();
        for(int i = 1; i < n; i++)
        {
            auto target = ans.back();
            auto s = folder[i].substr(0, target.size());
            if(target != s or folder[i][target.size()] != '/')
                ans.push_back(folder[i]);
        }  
        return ans;  
    }
};

File: 1233. Remove Sub-Folders from the Filesystem.py
class Solution:
    def isSubFolder(self, seen, folder_path, name=''):
        for ch in folder_path:
            if ch == '/' and name in seen:
                return True
            name += ch
        return name in seen
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        seen = set()
        folder.sort()
        for folder_path in folder:
            if not self.isSubFolder(seen, folder_path):
                seen.add(folder_path)
        return list(seen)